### ISSUE TITLE
`indent`: wrong indent when class property type is object

### ISSUE THREAD

**Issue by zzxming** [eslint-stylistic/eslint-stylistic#773]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

If class property have object type and object value, the indent will get wrong error.

```ts
// expect
class Some {
  attr: {
    index: number
    length: number
  } = {
    index: 0,
    length: 10
  }
}

// current rule expect
class Some {
  attr: {
    index: number
    length: number
  } = {
      index: 0,
      length: 10
    }
}

```

### Reproduction

https://stackblitz.com/edit/github-ppe14nzg?file=src%2Findex.ts

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests
